File: BackEnd-Siukat/services/value.go
package services

import (
	"BackEnd-Siukat/config"
	"BackEnd-Siukat/models"
	"time"
)

type ValueService struct{}

func (s *ValueService) GetByLoggedIn(noPeserta string) (models.Value, error) {
	db := config.DB
	var value models.Value

	err := db.Where("no_peserta = ?", noPeserta).First(&value).Error
	return value, err
}

func (s *ValueService) Add(v models.Value, atribut string) (models.Value, error) {
	db := config.DB

	v.Atribut = atribut

	err := db.Create(&v).Error
	return v, err
}

func (s *ValueService) AddLog(v models.Value, executor string, timestamp time.Time, atribut string) (models.LogValue, error) {
	db := config.DB

	if atribut == "" {
		atribut = "original"
	}

	logVal := models.LogValue{
		NoPeserta: v.NoPeserta,
		V1:        v.V1,
		V2:        v.V2,
		V3:        v.V3,
		V4:        v.V4,
		V5:        v.V5,
		Av1:       v.Av1,
		Bv2:       v.Bv2,
		Cv3:       v.Cv3,
		Dv4:       v.Dv4,
		Ev5:       v.Ev5,
		Ikb:       v.Ikb,
		Atribut:   atribut,
		Executor:  executor,
		Timestamp: timestamp,
	}

	err := db.Create(&logVal).Error
	return logVal, err
}

File: ZhammDataBeLama/siukat-service-dev_snmptn_2022/services/value.js
var Value = require('../models/value');
var LogValue = require('../models/log_value');

var value = function(){};

value.prototype.getByLoggedIn = function(no_peserta){
    return new Promise(function(resolve, reject){
      Value.findOne({
        where: {
          no_peserta: no_peserta
        }
      }).then(function(response){
        resolve(response)
      }).catch(function(err){
        reject(err)
      })
    })
}

value.prototype.add = function(users, atribut){
    return new Promise(function(resolve, reject){
      Value.create({
        no_peserta: users.no_peserta,
        v1: users.v1,
        v2: users.v2,
        v3: users.v3,
        v4: users.v4,
        v5: users.v5,
        av1: users.av1,
        bv2: users.bv2,
        cv3: users.cv3,
        dv4: users.dv4,
        ev5: users.ev5,
        ikb: users.ikb,
        atribut: atribut,
      }, {
        fields: [
            'no_peserta',
            'v1',
            'v2',
            'v3',
            'v4',
            'v5',
            'av1',
            'bv2',
            'cv3',
            'dv4',
            'ev5',
            'ikb',
            'atribut',
        ]
      }).then(function(response){
        resolve(response)
      }).catch(function(err){
        reject(err)
      })
    })
}

value.prototype.addLog = function(users, executor, timestamp, atribut="original"){
    return new Promise(function(resolve, reject){
      LogValue.create({
        no_peserta: users.no_peserta,
        v1: users.v1,
        v2: users.v2,
        v3: users.v3,
        v4: users.v4,
        v5: users.v5,
        av1: users.av1,
        bv2: users.bv2,
        cv3: users.cv3,
        dv4: users.dv4,
        ev5: users.ev5,
        ikb: users.ikb,
        atribut: atribut,
        executor: executor,
        timestamp: timestamp
      },{
        fields: [
          'no_peserta',
          'v1',
          'v2',
          'v3',
          'v4',
          'v5',
          'av1',
          'bv2',
          'cv3',
          'dv4',
          'ev5',
          'ikb',
          'atribut',
          'executor',
          'timestamp',
        ]
      }).then(function (response){
        resolve(response)
      }).catch(function (err) {
        reject(err)
      })
    })
  }

  module.exports = value